src/libraries/StateDiffDecoder.sol

```diff
@@ -0,0 +1,536 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {Vm, VmSafe} from "forge-std/Vm.sol";
+import {console} from "forge-std/console.sol";
+import {LibString} from "@solady/utils/LibString.sol";
+import {stdJson} from "forge-std/StdJson.sol";
+
+library StateDiffDecoder {
+    using LibString for string;
+    using stdJson for string;
+
+    address internal constant VM_ADDRESS = address(uint160(uint256(keccak256("hevm cheat code"))));
+    Vm internal constant vm = Vm(VM_ADDRESS);
+
+    struct StateDiff {
+        bytes32 slot;
+        bytes32 oldValue;
+        bytes32 newValue;
+    }
+
+    struct DecodedSlot {
+        string kind;
+        string oldValue; // Decoded stringified value.
+        string newValue; // Decoded stringified value.
+        string summary;
+        string detail;
+    }
+
+    struct DecodedStateDiff {
+        address who;
+        uint256 l2ChainId;
+        string contractName;
+        StateDiff raw;
+        DecodedSlot decoded;
+    }
+
+    // Temporary struct used during deduplication.
+    struct TempStateChange {
+        address who;
+        bytes32 slot;
+        bytes32 firstOld;
+        bytes32 lastNew;
+    }
+
+    // Leading underscore because some of the raw keys are reserved words in Solidity, and we need
+    // the keys to be ordered alphabetically here for foundry.
+    struct JsonStorageLayout {
+        string _bytes;
+        string _label;
+        uint256 _offset;
+        string _slot;
+        string _type;
+    }
+
+    // forgefmt: disable-start
+    bytes32 internal constant ERC1967_IMPL_SLOT = 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc;
```

Comment:
We've defined this same constant in Constants.sol in `contracts-bedrock/src/libraries`. Not sure if you want to import from there or not.

Reply:
will make this change in a follow up PR to avoid the need for re-approval here

Reply:
actually just going to leave this and avoid the `Constants.sol` import/dependency, since this is standardized